internal/tools/aws/database.go

```diff
@@ -228,14 +230,85 @@ func (d *RDSDatabase) rdsDatabaseProvision(installationID string, logger log.Fie
 
 		keyMetadata = enabledKeys[0]
 	} else {
-		keyMetadata, err = d.client.kmsCreateSymmetricKey(KMSKeyDescriptionRDS(awsID), []*kms.Tag{
+		caller, err := d.client.Service().sts.GetCallerIdentity(&sts.GetCallerIdentityInput{})
+		if err != nil {
+			return errors.Wrapf(err, "unable to get caller identity when creating RDS DB cluster %s", awsID)
+		}
+
+		callerARN, err := arn.Parse(*caller.Arn)
+		if err != nil {
+			return errors.Wrapf(err, "unable to parse caller ARN when creating RDS DB %s", awsID)
+		}
+
+		// This assumes that the root is at the same account of the caller.
+		rootARN := arn.ARN{
+			AccountID: callerARN.AccountID,
+			Service:   "iam",
+			Partition: "aws",
+			Resource:  "root",
+		}
+
+		// CMKs have their access restricted to users and actions defined in the key itself. Adding the
+		// root user to the policy prevents keys to become unmanageable in the future. In the policy
+		// level we need to give full permissions to the root principal so it can manage the key at
+		// any time if necessary. To other principals, we only allow actions necessary for performing
+		// a designated task. Note that the caller (or provisioner in this case) requires writting
+		// permissions to the key even if this actions are not defined in the user's IAM policy.
+		policyDocument := model.PolicyDocument{
+			Version: model.PolicyDocumentVersion,
+			Statement: []model.StatementEntry{
+				{
+					Sid:    "Allow full control of the key",
+					Effect: model.PolicyStatementEffectAllow,
+					Principal: map[string]string{
+						"AWS": rootARN.String(),
+					},
+					Action: []string{
+						"kms:*",
+					},
+					Resource: model.PolicyStatementAllResources,
+				},
```

Comment:
This part makes me a little nervous. I don't see anything inherently wrong with how you constructed the `rootARN` object above, but is there any way this could go wrong? Have you thought through this in regards to possibly giving permissions to an AWS user incorrectly?

I am only trying to channel the security team here and ensure we have done our best to not open the code up to new security issues. As I said, though, this seems like it should be fine as long as the `root` user will always be one or our cloud-team-owned accounts.

Reply:
> Have you thought through this in regards to possibly giving permissions to an AWS user incorrectly?

That's what AWS does if you don't provide a policy to the key.
https://docs.aws.amazon.com/kms/latest/developerguide/key-policies.html
https://aws.amazon.com/premiumsupport/knowledge-center/update-key-policy-future/

> As I said, though, this seems like it should be fine as long as the root user will always be one or our cloud-team-owned accounts.

Yes, that's a good point and that's why I left a comment when constructing the ARN that this assumes the caller and the root in the same AWS account.